source/globals.d.ts

```diff
@@ -64,8 +64,7 @@ interface GlobalEventHandlersEventMap {
 }
 
 declare namespace JSX {
-	type Element = Element | DocumentFragment;
```

Comment:
`type Element = Element |` was actually a loop and `JSX.Element` had become `any`

I wish TS threw an error on this

Reply:
I think here though, the `JSX.Element` type is a union type on the global `Element` (not itself) and `DocumentFragment` no?

Reply:
No, I think it was itself, hence the `any` that appeared. 

I saw React’s types create identical types outside the scope to avoid the problem, e.g.:

``` 
type NativeEvent = Event;
namespace React {
    type Event = BlahBlah<NativeEvent>
}

Reply:
It would be nice to get the types nailed down though for dom-chef. I'm just trying to avoid a rewrite of React types to accommodate our changes, but maybe that's what needs to be done. YOLO. 😉